overlord/aspectstate/aspectstate.go

```diff
@@ -49,11 +51,11 @@ func Set(st *state.State, account, bundleName, aspect, field string, value inter
 		return &aspects.AspectNotFoundError{Account: account, BundleName: bundleName, Aspect: aspect}
 	}
 
-	if err := asp.Set(databag, field, value); err != nil {
+	if err := asp.Set(tx, field, value); err != nil {
```

Comment:
this is nice, but I think we want one transaction at the level of api_aspects.go setAspect which likely requires changing the interface of aspectstate